### ISSUE TITLE
Minor difference in Validation between OHEA & OIHM

### ISSUE THREAD

**Issue by dbrownukk** [dbrownukk/EFD_v2#350]
Implemented the validation spreadsheet rules #340, but this has highlighted the fact that validation is now done in two places. (Four if we include both OHEA/OIHM)

1. During parse of spreadsheet
2. During 'Validate' from WGI or HH modules. 

I am thinking that they are both needed but need bringing together in one piece of code. Currently the parse validation checks RST and the Validation checks if asset/resource has status 'Invalid/Valid' and checks numeric values, as per #340 validation rule spreadsheet

**Comment by dbrownukk** [comment]
OHEA WGI Validation routine now uses the same code as the OIHM HH Validation. ValidationAsset.java 

Deployed

**Comment by daiClegg** [comment]
Instance: EFD_HM
App: OHEA 
Module: WGI Spreadsheet

Created Project, LZ, Community & WG.
Created Study (in the same Project & LZ - cool eh?)
In Community module, generated WGI template spreadsheet 
In Study module, generated HH template spreadsheet 

All testing below on this set up, unless documented otherwise

**Comment by daiClegg** [comment]
Anomaly 1
---
Filling in Spreadsheets in parallel, tab by tab...

-  Noticed that drop-downs for RST selection are not in the same order in the 2 spreadsheets.

This is true on at least Livestock, Other Tradable Goods & Food stocks.
Alphabetic in OIHM, no discernable order in OHEA (order entered in the RST table?)

However, this is not serious enough to warrant the time and effort to change at this point.

**Comment by daiClegg** [comment]
Anomaly 2
---
Filling in Spreadsheets in parallel, tab by tab...

In OHEA WGI Spreadsheet, on Assets -Food Stocks tab, the drop-down for RST selection is blank;
Can enter data but need the drop-down list for real-world use.

**Comment by dbrownukk** [comment]
Further thinking on this...

Now that the 'Validate' is done in one place fort WGI and HH I have added it to the parse -  it is now called after the parse of the WGI or HH interview spreadsheet. Seems sensible. 

The parse still does some validation of data entered into spreadsheet, but the Validate function is really the place to do all validation and let the parser do the parse...

**Comment by daiClegg** [comment]
**Anomaly 1 (reported above)**

Drop-downs for RST selection in WG Interview spreadsheet are not in the same order as HH interview spreadsheet.
This is true on at least Livestock, Other Tradable Goods & Food stocks.
Alphabetic in OIHM, no discernable order in OHEA (order entered in the RST table?)
Now alphabetic in both S/sheets

**Tested OK**

**Comment by daiClegg** [comment]
**Anomaly 2 (reported above)**

In OHEA WGI Spreadsheet, on Assets - Food Stocks tab, the drop-down for RST selection is blank;

**Tested OK**

**Comment by daiClegg** [comment]
Instance: EFD_HM
App:  OIHM 
Study: Test #340 
HH: 9

Tested upload of a "perfect" spreadsheet

**Tested OK**

**Comment by daiClegg** [comment]
Instance: EFD_HM
App: OHEA
Project: Test Common VAlidation #350
Village: Knowl Hill
WG: Comfortable

Uploaded the attached spreadsheet and parsed:

1. There was one information message - unparsed Food stocks. Checked it out and 'Apple' was invalid. Appears to be valid.
2. Checked Wild Foods and 'Jackfruit - wild' is marked valid but has no entry in the Name field.
3. Clicked Validate and only got a message for the Apple entry. Not for Jackfruit.

**Comment by daiClegg** [comment]
Next need to load the same content in HH Interview spreadsheet into OIHM...

**Comment by daiClegg** [comment]
Instance: EFD_HM
App: OIHM
Study: Test #350
Module: Household
Household no.: 1

The spreadsheet attached to this HH has exactly the same content (as far as i can see) as the one attached to:

App: OHEA
Project: Test Common VAlidation #350
Village: Knowl Hill
WG: Comfortable

But in OIHM it parses straight to Validate.

I redid the OHEA test in case anything significant had changed since i did the original test. But it still shows the same results.
 

Stu

**Comment by daiClegg** [comment]
I discovered why the difference:
 
Instance: EFD_HM

App: OHEA
LZProject: OHEA Training DemoUpload
LZ: Reading University
Village: Reading
WG: TestGroup1

the attached spreadsheet did not fully parse because:
'Apple' was not auto-validated as a Food Store.
And 'Jackfruit- wild' behaved the same as a Wild Food.

When i fixed the 2 manually, which was fine, Validation completed successfully.

APP: OIHM
Study: A Study
HH: 7 

The Spreadsheet attached to this HH has the same content as the OHEA s/sheet above for all tabs they have in common.
It does auto-parse 'Apple' & 'Jackfruit - wild' and goes immediately to Validated.

**Comment by daiClegg** [comment]
I think these differences in the validation are not material to the use of the apps, so i regard resolving this issue as very low priority

to be discussed, then probably IceBoxed.
